hushline/premium.py

```diff
@@ -0,0 +1,445 @@
+import asyncio
+import json
+from typing import Tuple
+
+import stripe
+from flask import (
+    Blueprint,
+    Flask,
+    current_app,
+    flash,
+    jsonify,
+    redirect,
+    render_template,
+    request,
+    session,
+    url_for,
+)
+from sqlalchemy import desc
+from werkzeug.wrappers.response import Response
+
+from .db import db
+from .model import StripeEvent, StripeInvoice, StripeInvoiceStatusEnum, Tier, User
+from .utils import authentication_required
+
+FREE_TIER = 1
+BUSINESS_TIER = 2
+
+
+def init_stripe() -> None:
+    stripe.api_key = current_app.config["STRIPE_SECRET_KEY"]
+
+
+def create_products_and_prices() -> None:
+    current_app.logger.info("Creating products and prices")
```

Comment:
With multiple containers, do we need additional guards to make this safe for concurrency?

Reply:
I've refactored this function so that it should be it's safe to run with multiple containers. When it runs it basically ensures  that the business tier's stripe_product_id and stripe_price_id match up with the Product and Price objects in Stripe. Running it multiple times shouldn't change anything, sort of like an ansible playbook.